src/main.rs

```diff
@@ -42,68 +63,56 @@ static GLOBAL_BASE: &str = "/home/sam/moods";
 static GLOBAL_DEST: &str = "/home/sam/moods-dest";
 
 #[get("/")]
-fn index(audio_mx: State<AudioMX>, flash: Option<FlashMessage>) -> Template {
-    let mp = audio_mx.mp.lock().expect("locking");
-    let dirs_vec = mp.keys().map(|x| x.to_string()).collect::<Vec<String>>();
-    match flash {
-        Some(m) => Template::render(
-            "index",
-            IndexCtx {
-                dirs: dirs_vec,
-                flash: Some(m.msg().to_string()),
-                flash_name: Some(m.name().to_string()),
-            },
-        ),
-        None => Template::render(
-            "index",
-            IndexCtx {
-                dirs: dirs_vec,
-                flash: None,
-                flash_name: None,
-            },
-        ),
-    }
+fn index(audio_mx: State<AudioMX>, flash_message: Option<FlashMessage>) -> Template {
+    let mp = audio_mx.lock().unwrap();
+    let dirs = mp.keys().map(Clone::clone).collect();
+    let flash = flash_message.as_ref().map(|m| m.msg().to_string());
+    let flash_name = flash_message.as_ref().map(|m| m.name().to_string());
+    Template::render(
+        "index",
+        IndexCtx {
+            dirs,
+            flash,
+            flash_name,
```

Comment:
If the field names are the same as local variable names, we can use `dirs,` as shorthand for `dirs: dirs,`.